external/js/kyber/src/curve/nist/curve.ts

```diff
@@ -0,0 +1,69 @@
+import { Group, Scalar, Point } from "../../index";
+import NistPoint from "./point";
+import NistScalar from "./scalar"
+import elliptic = require("elliptic");
+import BN = require("bn.js");
+
+export type BNType = string | Buffer | BN
+
+export default class Weierstrass implements Group {
+    curve: any;
+    redN: any;
+    bitSize: number
+    name: string
+    
+    constructor(config: { name: string, bitSize: number, gx: BNType, gy: BNType, p?: BNType, a?: BNType, b?: BNType, n: BN}) {
+        let { name, bitSize, gx, gy, ...options } = config;
+        this.name = name;
+        options["g"] = [new BN(gx, 16, "le"), new BN(gy, 16, "le")];
+        for (let k in options) {
+            if (k === "g") {
+                continue;
+            }
+            options[k] = new BN(options[k], 16, "le");
+        }
+        this.curve = new elliptic.curve.short(options);
+        this.bitSize = bitSize;
+        this.redN = BN.red(options.n);
+    }
+    
+    coordLen() {
```

Comment:
I'm just randomly browsing the code. How come some functions don't have a type? Like this one or `string()` and `inspect()` in some of the other files.